presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -29,11 +29,38 @@ ConfigBase::ConfigBase()
     : config_(std::make_unique<velox::core::MemConfig>()) {}
 
 void ConfigBase::initialize(const std::string& filePath) {
-  config_ = std::make_unique<velox::core::MemConfig>(
-      util::readConfig(fs::path(filePath)));
+  // See if we want to create a mutable config.
+  auto values = util::readConfig(fs::path(filePath));
+  bool mutableConfig{false};
+  auto it = values.find(std::string(SystemConfig::kMutableConfig));
+  if(it != values.end()) {
+    mutableConfig = folly::to<bool>(it->second);
+  }
+
+  if (mutableConfig) {
+    config_ = std::make_unique<velox::core::MemConfigMutable>(values);
+  } else {
+    config_ = std::make_unique<velox::core::MemConfig>(values);
+  };
+
   filePath_ = filePath;
 }
 
+folly::Optional<std::string> ConfigBase::setValue(
+    const std::string& propertyName,
+    const std::string& value) {
+  auto oldValue = config_->get(propertyName);
+  if (auto* memConfig =
+          dynamic_cast<velox::core::MemConfigMutable*>(config_.get())) {
+    memConfig->setValue(propertyName, value);
```

Comment:
maybe: return oldValue; here
and remove else

Reply:
Will all compilers be happy with missing return?
I'd rather leave it.